### ISSUE TITLE
Can't Build Demo

### ISSUE THREAD

**Issue by lawsonfulton** [BertrandBev/nlopt-js#2]
Hey there,

This looks like a cool project!

I'm trying to use it in a web-based react app. I'm having some trouble importing it, so I've been trying to build your demo to see if I can get it working.

However, yarn fails saying dependencies not found.

I notice in your `vue.config.js` there is the line
``` config.resolve.alias.set('@nlopt', path.resolve(__dirname, '../dist/index.js'))```

Do you happen to have any tips for how to go about creating this `dist/index.js`?

Thanks!

**Comment by MattFerraro** [comment]
Hi lawsonfulton,

I'm not the maintainer, but I went through a similar process in the last few days getting the demo to run. I have a modern working example [here](https://github.com/MattFerraro/constraint_solving) if you want to take a look.

My demo uses a vanilla create-react-app instance, then I just added whatever was needed to get nl_opt to import properly. That meant adding [config-overrides.js](https://github.com/MattFerraro/constraint_solving/blob/main/config-overrides.js) so I could inject a bunch of node dependencies. I had to separately npm install and then inject:
- assert
- buffer
- constants-browserify
- crypto-browserify
- graceful-fs (NOT fs!)
- path-browserify
- process
- stream-browserify

and of course `npm install react-app-rewired` and modify my build scripts to: `"start": "react-app-rewired start"` and `"build": "react-app-rewired build"`. Check out [package.json](https://github.com/MattFerraro/constraint_solving/blob/main/package.json) for the full working example.
